include/z64cutscene.h

```diff
@@ -737,18 +737,21 @@ typedef struct {
     /* 0xA */ s16 relativeTo; // see `CutsceneCamRelativeTo`
 } CsCmdCamPoint; // size = 0xC
 
-typedef enum {
-    /* 0 */ CS_CAM_INTERP_0,
-    /* 1 */ CS_CAM_INTERP_1,
-    /* 2 */ CS_CAM_INTERP_2,
-    /* 3 */ CS_CAM_INTERP_3,
-    /* 4 */ CS_CAM_INTERP_4,
-    /* 5 */ CS_CAM_INTERP_5,
-    /* 6 */ CS_CAM_INTERP_6,
-    /* 7 */ CS_CAM_INTERP_7
+typedef enum CutsceneCamInterpType {
+    /* 0 */ CS_CAM_INTERP_NONE, // values do not change.
+    // values 1-3 only uses a single point from the cmd
+    /* 1 */ CS_CAM_INTERP_SET, // values immediately set to cmd values.
+    /* 2 */ CS_CAM_INTERP_LERP, // values lerp from init to cmd values (increasing t)
+    /* 3 */ CS_CAM_INTERP_SMOOTH, // values lerp from cur to cmd values (fixed t at weight = 100)
```

Comment:
I think these two names have room for improvement, based on your comments the difference of note is that 2 moves from cmd's posA to posB, and 3 moves from current to cmd's pos?

Compared to that, 2 being linear and 3 being geometric seems less important 

Best (yet not good) I can think of atm is
```
2 POS_TO_POS
3 CUR_TO_POS
```

Unless I misunderstood your comments of course :^) gonna look at the code next

Reply:
Yeah, agree that the names could use improvement. If I understand the code correctly (and always open to verification/be corrected), the difference between these two are that: Imagine the camera Eye is interpolating from 0 to 10. (both of these only use 1 point, which is located at 10).

So `initPos` is 0, and `targetPos` is 10, and let's say 

Linear:
step 0) `lerp` = 0, `curPos` = 0
step 1) `lerp` = 0.25, `curPos` = 2.5
step 2) `lerp` = 0.50, `curPos` = 5.0
step 3) `lerp` = 0.75, `curPos` = 7.5
step 4) `lerp` = 1.0, `curPos` = 10

i.e. the `lerp` linearly increases with the ratio `curFrame` / `duration`. To get the `curPos`, there is a lerp betwen `initPos` and `targetPos` every time.


Smooth (lerp is fixed assuming weight is 100, which seems to be the default value?) so that it has more of a geometric lerp:
step 0) `lerp` = 0.5, `curPos` = 0.5
step 1) `lerp` = 0.5, `curPos` = 0.75
step 2) `lerp` = 0.5, `curPos` = 0.875
step 3) `lerp` = 0.5, `curPos` = 0.9375
step 4) `lerp` = 0.5, `curPos` = 0.96875
However, setting weight to something other than 100 means the lerp will slightly change as a function of `curFrame` but will still be relatively the same? A lot of rough assumptions here, please correct me if I'm wrong

I chose smooth since it reminded me of `Math_SmoothStepToS`, but I'm sure there is a proper name for this algorithm that I just don't know.

Reply:
For "smooth" is there also a
Step -1) pos = init pos from cmd
Or does it retain the previous position

The comment makes me think ot retains the previous position, the code makes me think that step -1) is applied

Reply:
There isn't a initial position it sets in any of these interpolations. It's more that the previous position of the camera is stored as the init in the interpolation struct, and retains that initial position throughout the duration
Step -1) init pos = pos:

Reply:
Oh ok
I got misled by / misunderstood the comments

Back to actual comments now that I think I understood correctly 

INTERP_LERP is redundant since LERP stands for linear interpolation

For naming the geometric ("discrete exponential") progression Smooth works but note SmoothStepToF is mostly linear and only becomes geometrical when nearing the target value
The camera code is full of "scale" / "scaleInv" parameters for geometrical stepping so maybe we could use "scale" ?

Also INTERP_SMOOTH isn't really an interpolation in the sense of a parameterized curve, the values are computed recursively

All in all maybe I'd suggest something like:
```
STEPTO_LINEAR // Lerp to the target position
STEPTO_SCALE // Step to the target position in increments of half the remaining distance
```
not 100% sold though

Reply:
I think it's a slight improvement to what I had, although `0.5f` was just an arbitrary number I pulled for the example, so reworded the comment since it's not "half"